File: No_3175/Find The First Player to win K Games in a Row.cpp
class Solution {
public:
    int findWinningPlayer(vector<int>& skills, int k) {
        int remaining = k, ret = 0, n = skills.size();

        for (int i = 1; i < n; i++) {
            if (skills[ret] > skills[i]) {
                remaining--;
            } else {
                ret = i;
                remaining = k - 1;
            }

            if (remaining == 0)
                return ret;
        }

        return ret;
    }
};

File: No_3175/Find The First Player to win K Games in a Row.py
class Solution:
    def findWinningPlayer(self, skills: List[int], k: int) -> int:
        ret = 0
        remaining = k

        for i, num in enumerate(skills[1:], start=1):
            if skills[ret] > num:
                remaining -= 1
            else:
                ret = i
                remaining = k - 1
            if remaining == 0:
                return ret

        return ret
        